packages/blade/src/components/Tabs/_decisions/decisions.md

```diff
@@ -0,0 +1,385 @@
+- [Filled Variant](#filled-variant)
+- [With Tooltip](#with-tooltip)
+- [Alternative APIs](#alternative-apis)
+  - [1. Quite tricky/hack to implement](#1-quite-trickyhack-to-implement)
+  - [2. The JSX doesn't map well with the visual hierarchy / the HTML structure](#2-the-jsx-doesnt-map-well-with-the-visual-hierarchy--the-html-structure)
+  - [3. Extensibility](#3-extensibility)
+  - [4. API Familiarity](#4-api-familiarity)
+- [Motion](#motion)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?node-id=3733%3A8981&mode=dev) to all variants of the Tabs component
+
+## Features
+
+- Controlled and Uncontrolled
+- Filled and Bordered variants
+- Horizontal and Vertical orientation
+- Medium and Large size
+
+## Anatomy
+
+<img src="./tabs-anatomy.png" width="100%" alt="Tabs thumbnail" />
+
+```jsx
+<Tabs>
+  <TabList>
+    <TabItem />
+  </TabList>
+  <TabPanel />
+</Tabs>
+```
+
+## `Tabs` API
+
+```ts
+type TabsProps = {
+  /**
+   * The content of the component, accepts `TabList` and `TabPanel` components.
+   */
+  children: React.ReactNode;
+  /**
+   * The value of the selected tab, If set the component will be controlled.
+   */
+  value?: string;
+  /**
+   * The default value of the selected tab, in case the Tabs component is uncontrolled.
+   */
+  defaultValue?: string;
+  /**
+   * Callback fired when the value changes.
+   */
+  onChange?: (value: string) => void;
+  /**
+   * The orientation of the tabs.
+   */
+  orientation?: 'horizontal' | 'vertical';
+  /**
+   * The size of the tabs.
+   */
+  size?: 'medium' | 'large';
+  /**
+   * The variant of the tabs.
+   */
+  variant?: 'bordered' | 'filled';
+  /**
+   * If `true`, the TabItems will grow to use all the available space.
+   */
+  autoWidth?: boolean;
+};
+```
+
+## `TabItem` API
+
+```ts
+type TabItemProps = {
+  /**
+   * The label of the tab item.
+   */
+  children: string;
+  /**
+   * The value of the tab item.
+   */
+  value: string;
+  /**
+   * Leading element of the tab item.
+   * Can be used to render an icon.
+   */
+  leading?: React.ReactNode;
+  /**
+   * Trailing element of the tab item.
+   * Can be used to render an badge/counter.
+   */
+  trailing?: React.ReactNode;
+  /**
+   * If `true`, the tab item will be disabled.
+   */
+  isDisabled?: boolean;
+};
+```
+
+## `TabPanel` API
+
+```ts
+type TabPanelProps = {
+  /**
+   * The value of the tab panel. This will be used to match the selected tab.
+   */
+  value: string;
+  /**
+   * The content of the tab panel.
+   */
+  children: React.ReactNode;
+};
+```
+
+### Examples:
+
+## Basic
+
+```js
+<Tabs>
+  <TabList>
+    <TabItem value="payments">Payments</TabItem>
+    <TabItem value="refunds">Refunds</TabItem>
+    <TabItem value="disputes">Disputes</TabItem>
+    <TabItem value="settlements">Settlements</TabItem>
+  </TabList>
+
+  <TabPanel value="payments">
```

Comment:
One con of this API is I'm not sure if we can make `value` type safe? For example, easy to mis-spell `value` in tab item and panel

Reply:
Yeah I think that we can't do anything. 

But should be fine, either way you should not have more than 7 or 8 tabs (maybe max 10), so don't foresee any maintainability issues. 

People can also do this

```
const paymentTabValue = "payments"

<TabItem value={paymentTabValue} />
<TabContent value={paymentTabValue} />
```